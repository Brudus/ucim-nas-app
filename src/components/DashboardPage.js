import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import filterWordsDue from '../selectors/words-due';

export const DashboardPage = ({ dueWordsCount }) => (
    <div>
        {dueWordsCount > 0 ? <Link to={`/study`}>Study now</Link> : <button disabled>Study now</button>}
    </div>
);

const mapStateToProps = (state) => ({
    dueWordsCount: filterWordsDue(state.words)
});

export default connect(mapStateToProps)(DashboardPage);