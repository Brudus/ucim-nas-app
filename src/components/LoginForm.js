import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export class LoginForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            loginError: '',
            emailPassed: undefined
        };
    }

    validateEmail = (email) => {
        const regex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
        return regex.test(email);
    };

    onEmailChange = (e) => {
        const email = e.target.value;
        const emailPassed = this.validateEmail(email);
        this.setState(() => ({ 
            email,
            emailPassed
        }));
    };

    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }));
    }

    onSubmit = (e) => {
        e.preventDefault();

        this.setState(() => ({ 
            loginError: '',
            emailPassed: true
        }));
        this.props.startLogin(this.state.email, this.state.password);
    };

    render() {
        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    {this.state.loginError && <p>{this.state.loginError}</p>}
                    <div>
                        <input 
                            type="email"
                            placeholder="Email address" 
                            autoFocus
                            onChange={this.onEmailChange}
                        />
                    </div>
                    
                    <div>
                        <input 
                            type="password"
                            placeholder="Password" 
                            onChange={this.onPasswordChange}
                        />
                    </div>
                    <button disabled={!this.state.emailPassed || !this.state.password}>Login</button>
                </form> 
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch, props) => ({
    startLogin: (email, password) => dispatch(startLogin(email, password))
});

export default connect(undefined, mapDispatchToProps)(LoginForm);