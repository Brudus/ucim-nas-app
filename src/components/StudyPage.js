import React from 'react';
import { connect } from 'react-redux';

export const StudyPage = (props) => (
    <div>
        <h2>Study!</h2>
        {props.words[0].source}
    </div>
);

const mapStateToProps = (state) => ({
    words: state.words
});

export default connect(mapStateToProps)(StudyPage);