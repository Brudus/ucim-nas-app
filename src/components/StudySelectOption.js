import React from 'react';

export default class StudySelectionOption extends React.Component {
    handleClick = () => {
        this.props.calculateNewSchedule(this.props.grade);
    };

    render() {
        return (
            <div>
                <button onClick={this.handleClick}>{this.props.text}</button>
            </div>
        );
    };
};