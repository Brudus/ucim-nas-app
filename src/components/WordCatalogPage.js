import React from 'react';
import WordList from './WordList';
import WordListFilters from './WordListFilters';
import WordsSummary from './WordsSummary';

const WordCatalogPage = () => (
    <div>
        <WordsSummary />
        <WordListFilters />
        <WordList />
    </div>
);

export default WordCatalogPage;