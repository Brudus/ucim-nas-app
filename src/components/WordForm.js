import React from 'react';

export default class WordForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            destination: props.word ? props.word.destination : '',
            source: props.word ? props.word.source : '',
            repeatAt: props.word ? props.word.repeatAt : undefined,
            repeatAtInverted: props.word ? props.word.repeatAtInverted : undefined,
            error: ''
        };
    }

    onDestinationChange = (e) => {
        const destination = e.target.value;
        this.setState(() => ({ destination }));
    };

    onSourceChange = (e) => {
        const source = e.target.value;
        this.setState(() => ({ source }));
    }

    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.destination || !this.state.source) {
            const error = 'Please provide a word and its translation in your language.'
            this.setState(() => ({ error }));
        } else {
            const error = '';
            this.setState(() => ({ error }));
            this.props.onSubmit({
                destination: this.state.destination,
                source: this.state.source,
                repeatAt: this.state.repeatAt ? this.state.repeatAt : Date.now(),
                repeatAtInverted: this.state.repeatAtInverted ? this.state.repeatAtInverted : Date.now()
            });
            this.setState(() => ({
                destination: '',
                source: '',
                repeatAt: undefined,
                repeatAtInverted: undefined
            }));
        }
    };

    render() {
        return (
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <input
                        type="text"
                        placeholder="Word in Serbo-Croatian"
                        autoFocus
                        value={this.state.destination}
                        onChange={this.onDestinationChange}
                    />
                    <input
                        type="text"
                        placeholder="Word in German"
                        value={this.state.source}
                        onChange={this.onSourceChange}
                    ></input>
                    <button>{ this.props.word ? 'Save' : 'Add' }</button>
                </form>
            </div>
        );
    };
}