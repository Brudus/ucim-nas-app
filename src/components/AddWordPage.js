import React from 'react';
import { connect } from 'react-redux';
import WordForm from './WordForm';
import { startAddWord } from '../actions/words';

export class AddWordPage extends React.Component {
    onSubmit = (word) => {
        this.props.startAddWord(word);
    };

    render() {
        return (
            <div>
                <h1>Add Word</h1>
                <WordForm 
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    };
}

const mapDispatchToProps = (dispatch) => ({
    startAddWord: (word) => dispatch(startAddWord(word))
});

export default connect(undefined, mapDispatchToProps)(AddWordPage);