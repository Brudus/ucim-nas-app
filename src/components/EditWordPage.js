import React from 'react';
import { connect } from 'react-redux';
import WordForm from './WordForm';
import { startEditWord, startRemoveWord } from '../actions/words';

export class EditWordPage extends React.Component {
    onSubmit = (word) => {
        this.props.startEditWord(this.props.word.id, word);
        this.props.history.push('/catalog');
    };
    onRemove = (id) => {
        this.props.startRemoveWord({ id: this.props.word.id });
        this.props.history.push('/catalog');
    };

    render() {
        return (
            <div>
                <WordForm 
                    word={this.props.word}
                    onSubmit={this.onSubmit}
                />
                <button onClick={this.onRemove}>Remove</button>
            </div>
        );
    };
};

const mapStateToProps = (state, props) => ({
    word: state.words.find((word) => word.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditWord: (id, word) => dispatch(startEditWord(id, word)),
    startRemoveWord: (data) => dispatch(startRemoveWord(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditWordPage);