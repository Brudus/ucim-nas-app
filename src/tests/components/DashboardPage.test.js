import React from 'react';
import { shallow } from 'enzyme';
import { DashboardPage } from '../../components/DashboardPage';

test('should render DashboardPage', () => {
    const dueWordsCount = 20;
    const wrapper = shallow(<DashboardPage dueWordsCount={dueWordsCount} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render DashboardPage without due words', () => {
    const dueWordsCount = 0;
    const wrapper = shallow(<DashboardPage dueWordsCount={dueWordsCount} />);
    expect(wrapper).toMatchSnapshot();
});