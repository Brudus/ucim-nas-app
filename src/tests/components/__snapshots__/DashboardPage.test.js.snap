// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashboardPage 1`] = `
<div>
  <Link
    to="/study"
  >
    Study now
  </Link>
</div>
`;

exports[`should render DashboardPage without due words 1`] = `
<div>
  <button
    disabled={true}
  >
    Study now
  </button>
</div>
`;
